### ISSUE TITLE
Topic create and alter

### ISSUE THREAD

**Issue by wangkewei111** [confluentinc/confluent-kafka-dotnet#329]
Description
===========
Topics are  automatically created when I use producer or consumer.
How to alter PartitionCount and ReplicationFactor?

**Comment by treziac** [comment]
You can't from client - you will have to use kafka-topics command.
An admin client is on the way (implemented in Kafka and available in Java) , but does not allow to change partition count and replication factor yet (only create, delete and alter config you can alter via kafka-topics among other operations) Creating and deleting are being added to librdkafka soon.

Also note that you can't create topics if topic autocreation is disabled on broker .

**Comment by treziac** [comment]
I may have misunderstand what you meant by alter. In kafka, `alter` typically means "modify after creation (which, as I stated, is not possible, 

If you want to create a topic with a given number of partition/ replica: it's not available in librdkafka/Confluent.Kafka yet, but it it's on roadmap, you can follow discussions here: https://github.com/edenhill/librdkafka/issues/986
For now, best and simplest way is to use `kafka-topics` (needing zookeeper endpoint and proper rights)

**Comment by mhowlett** [comment]
yep - use the `kafka-topics` command to create / alter topics. as @treziac notes, you can't use the client for this yet. you can turn off auto-topic creation with the `auto.create.topics.enable` broker config property (it's unfortunately not the default - I think that would be better).

**Comment by stefanloerwald** [comment]
What is the status of altering `ReplicationFactor` of a topic via C# API? Doesn't seem to be included in the API v1.2.1 and v1.3.0-pre, or am I missing something?

**Comment by mhowlett** [comment]
that's not supported by `AlterConfigs` (in any language)

**Comment by KrzysztofBranicki** [comment]
@mhowlett we stumbled on this as well. What is the reason why changing replication factor for existing topics is not supported (as you said in all languages)? Especially since you allow to change `min.insync.replicas` for existing topics. In our case this lead to a situation where we were able to change `min.insync.replicas` from 1 to 2, but we were not able change replication factor from 1 to 3 which would render topic unusable. `kafka-topics` CLI is not an option in our case as we want our microservices to manage their own topics in automated way.
